core/src/main/java/hudson/model/ItemGroupMixIn.java

```diff
@@ -170,12 +170,7 @@ public synchronized TopLevelItem createTopLevelItem( StaplerRequest req, Stapler
             String from = req.getParameter("from");
 
             // resolve a name to Item
-            Item src = null;
-            if (!from.startsWith("/"))
-                src = parent.getItem(from);
-            if (src==null)
-                src = Jenkins.getInstance().getItemByFullName(from);
-
+            Item src = Jenkins.getInstance().getItem(from, parent);
```

Comment:
Yes, hence the test. `../otherfolder/job` was suggested by completion but then did not actually work, since `getItem` can only accept a simple child name.